### ISSUE TITLE
Resolver-based discovery

### ISSUE THREAD

**Issue by jlink** [junit-team/junit5#107]
Depends on #106 being done first.

**Comment by bechte** [comment]
Also see #29.

**Comment by bechte** [comment]
Team decision:
- We choose to take pull request #218 instead of #134 to be able to ship M1 faster.
- Therefore, #134 will be closed and not merged
- I will open a new issue for extending the behaviour according to #134 that we can work on in a later milestone
